catkin_tools/context.py

```diff
@@ -474,14 +497,12 @@ def extend_path(self):
 
     @extend_path.setter
     def extend_path(self, value):
-        try:
-            if value is not None:
-                if not os.path.isabs(value):
-                    value = os.path.join(self.workspace, value)
-                get_resultspace_environment(value)
-            self.__extend_path = value
-        except IOError as exc:
-            raise ValueError("Unable to extend workspace from \"%s\": %s" % (value, exc.message))
+        if value is not None:
+            if not os.path.isabs(value):
+                value = os.path.join(self.workspace, value)
+            if not os.path.exists(value):
+                raise ValueError("Resultspace path '{0}' does not exist.".format(value))
+        self.__extend_path = value
```

Comment:
Why is the call to `get_resultspace_environement()` no longer needed here? Is it deferred to somewhere else?